google_guest_agent/diagnostics.go

```diff
@@ -116,13 +116,14 @@ func (d *diagnosticsMgr) set(ctx context.Context) error {
 		logger.Infof("Diagnostics: reject the request, as an existing process is collecting logs from the system")
 		return nil
 	}
-	cmd := exec.Command(diagnosticsCmd, args...)
+
 	go func() {
 		logger.Infof("Diagnostics: collecting logs from the system.")
-		out, err := cmd.CombinedOutput()
-		logger.Infof(string(out[:]))
-		if err != nil {
-			logger.Infof("Error collecting logs: %v", err)
+		res := run.WithOutput(ctx, diagnosticsCmd, args...)
+		logger.Infof(res.StdOut)
+		logger.Infof(res.StdErr)
```

Comment:
should we log `res.StdErr` only if `ExitCode != 0`?

Reply:
Previous code was using combined output so out is both stdout and stderr, I guess we want to expose a function to wrap combined as well and use it here - and be "more compatible" with the current implementation.

Reply:
done.